### PR TITLE
fix: [PIE-3848]: remove default infra type for serverless

### PR DESCRIPTION
**Pull request by satyam-harness**
##### Jira Links:
https://harness.atlassian.net/browse/PIE-3848

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Review by willd0yle — APPROVED** [review]

**Review by Tamaraius — APPROVED** [review]

File: src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraSpecifications.tsx [hunk 1]
```diff
@@ -183,10 +183,23 @@ export default function DeployInfraSpecifications(props: React.PropsWithChildren
     )
   }, [stage, getStageFromPipeline])
 
+  const infraGroups = React.useMemo(
+    () => getInfraGroups(selectedDeploymentType, getString, { NG_AZURE: defaultTo(NG_AZURE, false) }),
+    [selectedDeploymentType]
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraSpecifications.tsx:188]: NG_AZURE flag might be needed in dependency.

**Review by chetanvekariya-harness — DISMISSED** [review]

File: src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraSpecifications.tsx [hunk 2]
```diff
@@ -552,10 +565,14 @@ export default function DeployInfraSpecifications(props: React.PropsWithChildren
...
           </Accordion>
         ) : null}
-        <div className={stageCss.tabHeading} id="clusterDetails">
-          {defaultTo(detailsHeaderName[selectedInfrastructureType || ''], getString('cd.steps.common.clusterDetails'))}
-        </div>
-        <Card className={stageCss.sectionCard}>{getClusterConfigurationStep(selectedInfrastructureType || '')}</Card>
+        {selectedInfrastructureType && (
+          <>
+            <div className={stageCss.tabHeading} id="clusterDetails">
+              {getString('cd.steps.common.clusterDetails')}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraSpecifications.tsx:571]: Cluster Details will appear as header for every infra type. The detailsHeaderName was right thing to keep here.

Comment (satyam-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraSpecifications.tsx]: addressed

File: src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraSpecifications.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployInfraSpecifications/DeployInfraSpecifications.tsx]: addressed

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger coverage

**Pull request merged**